### PR TITLE
Unblock a User from Anomaly Detection (KEY-235)

### PR DESCRIPTION
**Pull request by zxan1285**
# ✏️ Changes
Added new user action - remove user blocks.
It removes all anomaly detection blocks from selected user.
`removeUserBlocksMenuItemText`, `removeBlocksDialogMessage` and `removeBlocksDialogTitle` props were added to the `languageDictionary` object.
`REMOVE_BLOCKS` was added to the list of error translations.

# 📷 Screenshots
 

# 🔗 References
Jira: https://auth0team.atlassian.net/browse/KEY-235

# 🎯 Testing
✅ This has been tested locally
✅ This has been tested in a webtask
✅ This change has unit tests

### CODE REVIEW

**Comment by auth0extensions** [comment]
MD5: 74da4121dc1c0ed2a8e5b0741f824034

# NSP - Security Report

_This is an automated security audit of this project. Please do not modify its contents_

Vulnerabilities found: **11**
Affected Modules: **superagent, superagent, mime, mime, debug, fresh, hoek, lodash, lodash, safe-eval, superagent**
Summary:

- Large gzip Denial of Service
- Large gzip Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Prototype pollution attack
- Prototype Pollution
- Prototype Pollution
- Sandbox Breakout
- Large gzip Denial of Service


## Detailed report
#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-delegated-admin@3.1.0 > auth0-extension-express-tools@1.1.8 > auth0-extension-tools@1.3.3 > superagent@2.3.0**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-delegated-admin@3.1.0 > auth0-extension-tools@1.3.3 > superagent@2.3.0**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-delegated-admin@3.1.0 > express@4.14.0 > send@0.14.1 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-delegated-admin@3.1.0 > superagent@1.2.0 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **debug**
Installed version: **2.2.0**
Patched versions: **>= 2.6.9 < 3.0.0 || >= 3.1.0**
Advisory: https://nodesecurity.io/advisories/534
Path: **auth0-delegated-admin@3.1.0 > express@4.14.0 > debug@2.2.0**

---

#### Regular Expression Denial of Service

Affected Module: **fresh**
Installed version: **0.3.0**
Patched versions: **>= 0.5.2**
Advisory: https://nodesecurity.io/advisories/526
Path: **auth0-delegated-admin@3.1.0 > express@4.14.0 > fresh@0.3.0**

---

#### Prototype pollution attack

Affected Module: **hoek**
Installed version: **2.16.3**
Patched versions: **> 4.2.0 < 5.0.0 || >= 5.0.3**
Advisory: https://nodesecurity.io/advisories/566
Path: **auth0-delegated-admin@3.1.0 > jsonwebtoken@7.4.3 > joi@6.10.1 > hoek@2.16.3**

---

#### Prototype Pollution

Affected Module: **lodash**
Installed version: **3.10.1**
Patched versions: **>=4.17.5**
Advisory: https://nodesecurity.io/advisories/577
Path: **auth0-delegated-admin@3.1.0 > lodash@3.10.1**

---

#### Prototype Pollution

Affected Module: **lodash**
Installed version: **4.5.1**
Patched versions: **>=4.17.5**
Advisory: https://nodesecurity.io/advisories/577
Path: **auth0-delegated-admin@3.1.0 > lru-memoizer@1.10.0 > lodash@4.5.1**

---

#### Sandbox Breakout

Affected Module: **safe-eval**
Installed version: **0.3.0**
Patched versions: **<0.0.0**
Advisory: https://nodesecurity.io/advisories/337
Path: **auth0-delegated-admin@3.1.0 > safe-eval@0.3.0**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.2.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-delegated-admin@3.1.0 > superagent@1.2.0**

---

File: client/utils/getErrorMessage.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ const defaultMessages = {
   UPDATE_SCRIPT: { default: 'An error occurred while saving the script: {message}' },
   FETCH_SETTINGS: { default: 'An error occurred while loading the settings: {message}' },
   UNBLOCK_USER: { default: 'An error occurred while unblocking the user: {message}' },
+  REMOVE_BLOCKS: { default: 'An error occurred while unblocking the user: {message}' },
```
Comment (fyockm) [inline client/utils/getErrorMessage.js]: to differentiate, should we change this to 'An error occurred while removing anomaly blocks:`

**Comment by zxan1285** [comment]
We can rename methods as we want, but i'm not sure how exactly they should be named. Right now, `unblockUser` stands for removing `blocked` flag from user object, and `removeBlocks` - is for removing all the `user-blocks`.

**Comment by FadyMak** [comment]
@zxan1285 Sorry to create more work but why we are we not mimicking the behaviour of Manage and combining it into a single action that performs 2 calls on the backend? In this case we would have a single `Block User` vs `Unblock User` action and there's no need to treat removing the user blocks differently from unblocking the user.

**Comment by zxan1285** [comment]
@FadyMak but it is working in that way on the Dashboard.
Here:
https://www.screencast.com/t/KJTvycNO
https://www.screencast.com/t/fDHwoD20T

**Comment by FadyMak** [comment]
You're right @zxan1285 - sorry about that. I was under the impression it was a single action.

**Comment by fyockm** [comment]
@zxan1285 “This endpoint can be used to retrieve a list of blocked IP addresses of a particular user given a user_id.”

so can we rename that `removeBlocks` method `removeBlockedIPs` or something like that?

**Pull request merged**